app/models/comment.rb

```diff
@@ -65,19 +63,19 @@ def akismet_attributes
   after_create :update_work_stats
   after_destroy :update_work_stats
 
+  before_update :moderate_update
   after_update :after_update
+
+  # Put comments back into moderation after significant changes
+  def moderate_update
+    return if !moderated_commenting_enabled? || is_creator_comment?
+    self.unreviewed = true if significant_content_change?
+  end
+
   def after_update
     users = []
     admins = []
 
-    if self.saved_change_to_edited_at? && self.saved_change_to_comment_content? && self.moderated_commenting_enabled? && !self.is_creator_comment?
```

Comment:
Why did you stop checking whether `edited_at` has changed?

(While I do think the rewritten code is cleaner, it makes it harder to see how the code has changed, and what the issue was: `self.comment_content_was` vs `self.comment_content_before_last_save`.)

Reply:
I couldn't think of a scenario where that would matter, but now I'm pretty sure it's so we don't catch comments marked deleted (which have their content changed). I'll put it back and make a note to make that a little more explicit.

Reply:
I think it might also matter if the sanitizer makes significant changes. (Say, if you disallow an HTML tag that used to be permitted.)